### PR TITLE
Fix date formatting

### LINKED ISSUE #499: Date Formatting Wrong
image

I tried using the YYYY-MM-DD , MM/DD/YYYY AND DD/MM/YYYY and they all format wrong.

### PR DESCRIPTION
**Pull request by lisa-fehr**
Closes #499

### CODE REVIEW

**Comment by lisa-fehr** [comment]
`vue-form-elements/src/dateUtils.js@getUserDateTimeFormat`
- Needs an update to add time to the `datetime_format`, if the user selects a date only time format in user settings (Y/m/d). 
- Or those options need to be removed from the date format selection list.

File: src/components/renderer/form-masked-input.vue [hunk 1]
```diff
@@ -40,6 +40,8 @@ import ValidationMixin from '@processmaker/vue-form-elements/src/components/mixi
 import DataFormatMixin from '@processmaker/vue-form-elements/src/components/mixins/DataFormat';
 import Inputmasked from './form-input-masked';
 import { TheMask } from 'vue-the-mask';
+import { getUserDateFormat, getUserDateTimeFormat } from '@processmaker/vue-form-elements/src/dateUtils';
```
Comment (chipit24) [inline src/components/renderer/form-masked-input.vue]: I wonder if it's worth exposing these functions as part of what `@processmaker/vue-form-elements` exports?

Although this doesn't seem cause any issues, babel by default ignores imports from `node_modules`, so these imports won't actually be transpiled by babel, but we're not using any fancy new JavaScript features so there's no issues.

**Comment by chipit24** [comment]
> * Or those date only options need to be removed from the date format selection list.

I think this is what should happen. @velkymx can confirm.

File: src/components/renderer/form-masked-input.vue [hunk 2]
```diff
@@ -72,14 +101,24 @@ export default {
...
     convertFromData(value) {
       if (this.dataFormat === 'percentage') return value * 100;
+      if (this.dataFormat === 'datetime') return moment(value).format(this.getUserDateTimeFormat());
+      if (this.dataFormat === 'date') return moment(value).format(this.getUserDateFormat());
       return value;
     },
+    getMask() {
+      const format = this.dataFormat === 'date' ? this.getUserDateFormat() : this.getUserDateTimeFormat();
+      return typeof masks[format] === 'undefined'
+        ? masks['default']
```
[the remark this answers is not in the document]
Comment (lisa-fehr) [reply src/components/renderer/form-masked-input.vue]: Weird. This wasn't the most up-to-date code. I switched this to `defaultMask()`

**Pull request merged**